jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java

```diff
@@ -593,13 +593,18 @@ private static class ServletCoreRequest extends Request.Wrapper
             String includedServletPath = (String)request.getAttribute(RequestDispatcher.INCLUDE_SERVLET_PATH);
             boolean included = includedServletPath != null;
             if (request.getDispatcherType() == DispatcherType.REQUEST)
-                _uri = getWrapped().getHttpURI();
+                _uri = Request.newHttpURIFrom(getWrapped(), encodedPathInContext);
```

Comment:
I'm now cautious about us rewriting the URI.  We should just reconstruct it from servletPath and pathInfo, but not change it for pathInfoOnly.